vars/maven.groovy

```diff
@@ -1,16 +1,21 @@
 import java.util.Properties
 
-def runMavenWithSettings(String settingsXmlId, String goals, Properties properties, String logFileName = null) {
-    configFileProvider([configFile(fileId: settingsXmlId, variable: 'MAVEN_SETTINGS_XML')]) {
-        def propertiesString = ''
-
+def runMaven(String goals, List options=[], Properties properties=null, String logFileName = null) {
+    def mvnCommand = "mvn -B ${options.size() > 0 ? options.join(' ') + ' ' : ''}${goals}"
```

Comment:
I think the ternary operation is redundant for the join() on an empty list should result in an empty string.

Reply:
updated `runMaven`. Please have a look